api/wasm/v0/cpp/proxy_wasm_externs.h

```diff
@@ -0,0 +1,126 @@
+/*
```

Comment:
High level question: So far `api/` is about configuration/proto APIs. Is this where we want the WASM API to live? Would it make more sense to put it in `include/envoy/wasm` or `include/wasm`? @htuch WDYT?

Reply:
Since these files are not normally included in Envoy code proper it might be confusing to put it in include/.  Perhaps include/api/wasm ?

Reply:
Sure sounds good, let's do that.

Reply:
Yeah, I guess the question is what parts of this are API, what parts are ABI, and what parts are the Envoy interfaces. In general, I think it would be least confusing to keep code interfaces outside of `api/`.

Reply:
The SDK includes .cc files, so it wouldn't make sense to put those under include/.

Reply:
This would mean separating the API/SDK and the ABI where the ABI could be under include/api/wasm (which is now a bit confusing since it is not the API but instead the ABI).

We also now will need a place to put the API/SDK.  sdk/ ?

Reply:
So how about: 
include/abi/wasm: ABI
sdk/wasm/: Various language SDKs?

Reply:
(I think also include/envoy/abi/wasm would be fine also for the ABI FWIW)